internal/handshake/retry.go

```diff
@@ -0,0 +1,42 @@
+package handshake
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+)
+
+var retryAEAD cipher.AEAD
```

Comment:
cipher.AEAD doesn't seem to make any thread safety guarantees. I think you might need a mutex here, or make this a server local var?

Reply:
I'm not sure I understand what you mean. `retryAEAD` is only set in `init()`, after we just call `Seal`.
`Seal` doesn't modify any internal state (at least not that I'm aware of), so it should be safe to use it concurrently, right?

Reply:
As long as that’s not guaranteed by the interface or the constructor, you can’t make that assumption.